src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMEnvironmentContributor.java

```diff
@@ -0,0 +1,89 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.google.common.annotations.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.*;
+import hudson.plugins.git.GitSCM;
+import hudson.scm.SCM;
+import jenkins.branch.BranchSource;
+import jenkins.branch.MultiBranchProject;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import java.io.IOException;
+import java.util.Objects;
+
+@Extension
+public class BitbucketSCMEnvironmentContributor extends EnvironmentContributor {
+
+    public void buildEnvironmentFor(@NonNull Job job, @NonNull EnvVars envs, @NonNull TaskListener listener) throws IOException, InterruptedException {
+        if (job instanceof AbstractProject) {
```

Comment:
I counted 14 invocations of this method for my one Pipeline job. I wonder if this now becomes a performance concern? I _think_ it is fine, it is a lot of code but most of it is not executing, so I guess we're ok.